Mathlib/Algebra/Lie/Semisimple.lean

```diff
@@ -49,56 +52,82 @@ class IsSimple extends LieModule.IsIrreducible R L L : Prop where
   non_abelian : ¬IsLieAbelian L
 #align lie_algebra.is_simple LieAlgebra.IsSimple
 
-/-- A semisimple Lie algebra is one with trivial radical.
+/--
+A Lie algebra *has trivial radical* if its radical is trivial.
+This is equivalent to having no non-trivial solvable ideals,
+and further equivalent to having no non-trivial abelian ideals.
+
+In characteristic zero, it is also equivalent to `LieAlgebra.IsSemisimple`.
 
 Note that the label 'semisimple' is apparently not universally agreed
 [upon](https://mathoverflow.net/questions/149391/on-radicals-of-a-lie-algebra#comment383669_149391)
-for general coefficients. We are following [Seligman, page 15](seligman1967) and using the label
-for the weakest of the various properties which are all equivalent over a field of characteristic
-zero. -/
+for general coefficients.
+
+For example [Seligman, page 15](seligman1967) uses the label for `LieAlgebra.HasTrivialRadical`,
+whereas we reserve it for Lie algebras that are a direct sum of simple Lie algebras.
+-/
+class HasTrivialRadical : Prop where
+  trivial_radical : radical R L = ⊥
+#align lie_algebra.is_semisimple LieAlgebra.HasTrivialRadical
+
+/--
+A *semisimple* Lie algebra is one that is a direct sum of simple Lie algebras.
+
+Note that the label 'semisimple' is apparently not universally agreed
+[upon](https://mathoverflow.net/questions/149391/on-radicals-of-a-lie-algebra#comment383669_149391)
+for general coefficients.
+
+For example [Seligman, page 15](seligman1967) uses the label for `LieAlgebra.HasTrivialRadical`,
+the weakest of the various properties which are all equivalent over a field of characteristic zero.
+-/
 class IsSemisimple : Prop where
-  semisimple : radical R L = ⊥
-#align lie_algebra.is_semisimple LieAlgebra.IsSemisimple
+  spanning : sSup {I : LieIdeal R L | IsAtom I} = ⊤
```

Comment:
According to [rule 7](https://leanprover-community.github.io/contribute/naming.html#general-conventions) the usual conventions apply to constructor fields too:
```suggestion
  sSup_isAtom_eq_top : sSup {I : LieIdeal R L | IsAtom I} = ⊤
```
(and similarly the other two fields)

Reply:
The last field was already fine, no?